### ISSUE TITLE
[Bug] Undefined method collate_by - Ubuntu - ffi-icu and icu library

### ISSUE THREAD

**Issue by Kulgar** [instructure/canvas-lms#711]
Hello :)

I ran into this exact same problem: https://groups.google.com/forum/#!topic/canvas-lms-users/SR_mmZIzge0

In Ubuntu 12 or 14 TLS, I found that we have to install "libicu-dev", so I did as usual: 

```bash
sudo apt-get install libicu-dev
```

But then we have the following error (that I found by debugging the whole thing, removing the begin/rescue in lib/canvas/icu.rb): 

```ruby
> ICU::Lib.attach_function(:ucol_getSortKey, "ucol_getSortKey#{suffix}", [:pointer, :pointer, :int, :pointer, :int], :int)
FFI::NotFoundError: Function 'ucol_getSortKey' not found in [/usr/lib/x86_64-linux-gnu/libicui18n.so.52, /usr/lib/x86_64-linux-gnu/libicutu.so.52]
	from /usr/local/rvm/gems/ruby-1.9.3-p551@canvas/gems/ffi-1.1.5/lib/ffi/library.rb:249:in `attach_function'
	from (irb):7
	from /usr/local/rvm/gems/ruby-1.9.3-p551@canvas/gems/railties-3.2.19/lib/rails/commands/console.rb:47:in `start'
	from /usr/local/rvm/gems/ruby-1.9.3-p551@canvas/gems/railties-3.2.19/lib/rails/commands/console.rb:8:in `start'
	from /usr/local/rvm/gems/ruby-1.9.3-p551@canvas/gems/railties-3.2.19/lib/rails/commands.rb:41:in `<top (required)>'
	from script/rails:6:in `require'
	from script/rails:6:in `<main>'
```

The problem lies in the ffi-icu library that can't find the right "suffix" for the Ubuntu ICU library. I have made a pull request here: https://github.com/jarib/ffi-icu/pull/30 and explained the issue here: https://github.com/jarib/ffi-icu/issues/29

If the pull request is merged, I would like to update ffi-icu library used in Canvas LMS. Would that be ok? 

Until then I will use my version of the gem in my Canvas applications (v0.1.9) from my fork. So I'll be able to see if I run into any other problems.

Thanks!
Regis

**Comment by simonista** [comment]
Hey @Kulgar, I'm not super familiar with libicu, but that all sounds reasonable to me, we'll keep an eye out for the pull request.  The part I don't understand, though, is that I believe we run on Ubuntu 12 TLS without issue.  @ccutrer, any thoughts here?

**Comment by Kulgar** [comment]
@simonista : I don't think it crashes the application as it fall back to a "Naive" Collator if it can't find the right icu lib if I understood well the code. 

Also since I posted my issue,  I tried running the spec "spec/lib/canvas/icu_spec.rb" and they almost all fail for me. I'm using last Canvas version and Ubuntu 14.04. I had to fix some things there even after reparing lib-icu. I'll include my fix in my PR.
